contracts/StorageAccessible.sol

```diff
@@ -0,0 +1,26 @@
+pragma solidity ^0.5.2;
+
+
+/// @title StorageAccessible - generic base contract that allows callers to access all internal storage.
+contract StorageAccessible {
+    /**
+     * @dev Reads `length` bytes of storage in the currents contract\\
+     * @param offset - the offset in the current contract's storage in words to start reading from
+     * @param length - the number of words (32 bytes) of data to read
+     * @return the bytes that were read.
+     */
+    function getStorageAt(uint256 offset, uint256 length)
+        public
+        view
+        returns (bytes memory)
+    {
+        bytes memory result = new bytes(length * 32);
+        for (uint256 index = 0; index < length; index++) {
+            assembly {
+                let word := sload(add(offset, index))
+                mstore(add(add(result, 0x20), mul(index, 0x20)), word)
```

Comment:
Solidity assembly note: I find it super confusing that `sload` takes the location in "words" and `mstore` take the location in bytes .... who came up with that .... On that note I like that you normalized it to "words" for the method, but feels a little less intuitive (still the best way).